### PR TITLE
Feature/code rag

### PR DESCRIPTION
**Pull request by cgrant**
adding rag load and query

### CODE REVIEW

File: devai-cli/src/devai/commands/rag/load.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+from pathlib import Path
+from git import Repo
+
+from langchain_community.document_loaders import GitLoader
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_google_vertexai import VertexAIEmbeddings
+from langchain_community.vectorstores import Chroma
+
+
+def load_docs(repo_path, branch, local_dir, allowed_extensions):
```
Comment (gitrey) [inline devai-cli/src/devai/commands/rag/load.py:14]: this load everything in the repo that is allowlisted in the extensions list, right? not just docs.

File: devai-cli/src/requirements.txt [hunk 2]
```diff
@@ -4,4 +4,27 @@ jira==3.6.0
...
 langchain_google_vertexai==0.0.5
-atlassian-python-api==3.41.10
\ No newline at end of file
+atlassian-python-api==3.41.10
+
+#dotenv
+
+
+
+google-cloud-aiplatform
```
Comment (gitrey) [inline devai-cli/src/requirements.txt:13]: some of these already present with versions info.

can you add versions for rest of them?

**Review by gitrey — APPROVED** [review]
LGTM

File: devai-cli/src/devai/commands/rag/load.py (hunk 1 above)
Comment (cgrant) [reply devai-cli/src/devai/commands/rag/load.py:14]: correct

**Pull request merged**